Parent/Parent/Profile/ProfilePresenter.swift

```diff
@@ -19,52 +19,111 @@ import Core
 import CanvasKeymaster
 
 public class ProfilePresenter: ProfilePresenterProtocol {
+    let env: AppEnvironment
+    public weak var view: ProfileViewControllerProtocol?
+
     #if DEBUG
     var showDevMenu = true
     #else
     var showDevMenu = UserDefaults.standard.bool(forKey: "showDevMenu")
     #endif
 
-    public weak var view: ProfileViewController?
+    lazy var permissions: Store<GetContextPermissions> = {
+        let useCase = GetContextPermissions(context: ContextModel(.account, id: "self"), permissions: [.becomeUser])
+        return env.subscribe(useCase) { [weak self] in
+            self?.view?.reload()
+        }
+    }()
+
+    var canMasquerade: Bool {
+        if env.currentSession?.actAsUserID != nil {
+            return false
+        }
+
+        if env.api.baseURL.absoluteString.contains("siteadmin.instructure.com") {
+            return true
+        }
+
+        guard let becomeUser = self.permissions.first?.becomeUser else {
+            return false
+        }
+        return becomeUser
+    }
+
     public var cells: [ProfileViewCell] {
-        return [
+        var cells: [ProfileViewCell] = []
+
+        if self.canMasquerade {
```

Comment:
I like these conditions much better than the `hidden` flag from before. Good improvement.